addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -1759,3 +1759,34 @@ def test_multi_users_enabled(self):
         self.assertEqual(res_pick.move_line_ids, expected_backorder_mls)
         # Original picking in state done
         self.assertEqual(pick.state, "done")
+
+
+class TestStockPickingPriorities(common.BaseUDES):
+    def test_priority_remains_on_done_picking(self):
+        """
+        Test that priorities are not overwritten on done pickings.
+        """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 100)
+        pick = self.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{"product": self.fig, "uom_qty": 10}],
+            location_id=self.test_stock_location_02.id,
+            priority="1",
```

Comment:
There's a case for testing both `priority="1"` _and _ `priority="0"`, as otherwise the tests pass if we just hardcode the priority to "1" in the implementation.  What do you think @PeterC10 ?

Reply:
I may be misunderstand this, but the problem was it was resetting the priority back to 0. Not the other way around, so not sure how testing if priority is still 0 is beneficial?

Reply:
I see your point... gets tricky because by default Odoo sets it to zero anyway.

Better to be safe though, so yes let's have a picking for each priority, and assert that it doesn't change when completed or cancelled.

Will be more useful in future when we implement additional priorities.

Reply:
Then we can be confident that the system is behaving as expected in both cases.